### DISCUSSION TITLE
How to change a value in a javascript file? Help question!

### DISCUSSION THREAD

**Discussion by Alex-e107nl** [e107inc/e107#5376] [General]
In the file admin.jquery.js is the delay value of the tooltip, I'm getting a bit nervous about the settings now, it comes too fast and hangs too long... now I can of course do a pull with the change but I would like to adjust the value of the delay from the admin_theme.php, how can I do that? Has anyone done that before? The 'show' is now 0 but with 600 it is much more relaxing...

I still need to adjust some more javascript and it would be nice if I could get some help with this...

```
$this.tooltip({
					title: function ()
					{
						return $fieldHelp.html();
					},
					fade: true,
					html: true,
					opacity: 1.0,
			//		trigger: 'hover focus',
					placement: placement,
					container: 'body',
					delay: { 
						show: 0,
						hide: 600
					}
				});
```

**Comment by Jimmi08** [comment]
Maybe I am wrong...   maybe you will get the correct answer or at least they confirm this one.  

 It has 2 steps.  First, add settings to your theme, like this:  

```
$options['tooltip']['show'] = '500';  //example
$jsSettings = array(
	'options' => $options['tooltip']
);

e107::js('settings', array('fieldHelpTooltip' => array($jsSettings)));
```

Look for selectize in form_handler.php to see this already in action. 

I used 'fieldHelpTooltip'  only to be easily found in the source code (tooltip is there many times).  Then check the source code, it should look like this:
![image](https://github.com/user-attachments/assets/3d0dc81b-295f-4fdb-924f-ddc6608e0f87)

Now you have it available for any script in e107 environment if it is supported.   

To get it supported is another step.  Coming soon...

**Reply by Jimmi08** [reply]
Ha, it has to be named this way because of this:  `e107.behaviors.fieldHelpTooltip = {` in admin.jquery.js

Use it here: 
show: e107.settings.fieldHelpTooltip.XY  - I am not familiar with javascript, so I cannot get this value from $settings

But you can use console.log($settings) to see the structure and use it. 

This is maybe a better idea:
```
$options['show'] = '500';
$jsSettings = array(
	'options' => $options
);
```
Or call that bootstrap5 option to have more settings there. 

![image](https://github.com/user-attachments/assets/da499320-dfd9-4f08-bd67-89d01c6f600d)
 
Just look for e107.settings.  in the code to get it right.  And don't forget to add this to bootstrap3 with value 0...  Or set the default value for these settings somewhere... 

EDIT:  working code with that second setting, thanks GPT:
```
					delay: {
                        show: e107.settings.fieldHelpTooltip[0].options.show,
						hide: 600
					}
```

**Reply by Alex-e107nl** [reply]
Thx for helping! @Jimmi08 but i just cannot get it working with a inline script, the only way to solve this for now is calling an external .js file from admin_theme.php `e107::js('theme', "js/admin/delaytooltip.js", 'jquery');` with the code copied from admin.jquery.js...

When i release the theme there wil be (i hope) someone who knows a better way :-)

(ps, and yes there are 2.....e-tip and tooltip...)

```
$(document).ready(function()
{


	$(".e-tip").each(function() {
						
			
			var tip = $(this).attr('title');
			if(!tip)
			{
				return;
			}
			
			var pos = $(this).attr('data-placement'); 
			if(!pos)
			{
				pos = 'top';
			}
			
			$(this).tooltip({opacity:1.0,fade:true, placement: pos, delay: {show: 500, hide: 100}});
			// $(this).css( 'cursor', 'pointer' )
		});
		
		
	e107.behaviors.fieldHelpTooltip = {
		attach: function (context, settings)
		{
			var selector = '.admin-ui-help-tip,div.tbox,div.checkboxes,input,textarea,select,.e-tip,div.form-control';

			$(context).find(selector).once('field-help-tooltip').each(function ()
			{
				var $this = $(this);
				var $fieldHelp = $this.nextAll(".field-help");
				var placement = 'bottom';

				if($this.is("textarea"))
				{
					placement = 'top';
				}

				// custom position defined in field-help container class
				var custPlace = $fieldHelp.attr('data-placement'); // ie top|left|bottom|right
				if(custPlace !== undefined)
				{
					placement = custPlace;
				}

				// custom position defined in selector tag.
				var pos = $(this).attr('data-tooltip-position');
				if(pos !== undefined)
				{
					placement = pos;
				}
				

				$fieldHelp.hide();

				$this.tooltip({
					title: function ()
					{
						return $fieldHelp.html();
					},
					fade: true,
					html: true,
					opacity: 1.0,
			//		trigger: 'hover focus',
					placement: placement,
					container: 'body',
					delay: { 
						show: 500,
						hide: 100
					}
				});
			});
		}
	};
});
```

**Reply by Alex-e107nl** [reply]
Oh sorry, you need to slow down 3 tooltip configurations.... missed this one :-)

```
var tooltipTriggerList = [].slice.call(document.querySelectorAll('[data-bs-toggle="tooltip"]'))
	var tooltipList = tooltipTriggerList.map(function(tooltipTriggerEl) {
	return new bootstrap.Tooltip(tooltipTriggerEl,{delay: { "show": 600, "hide": 100 }});
	})	
```
